src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,207 +1,850 @@
+        r = hypot(pt.x, pt.y) 
+        r <= c.r || csg_isapprox_lin(r, c.r)
+    end &&
+    _in_angular_interval_closed(atan(pt.y, pt.x), c.φ)
+end
+function _in(pt::CartesianPoint, c::PartialCylinder{T,OpenPrimitive}) where {T} 
+    abs(pt.z) < c.hZ &&
+    hypot(pt.x, pt.y) < c.r &&
+    _in_angular_interval_open(atan(pt.y, pt.x), c.φ)
 end
-Cone(rbotMin, rbotMax, rtopMin, rtopMax, φMin, φMax, zMin, zMax) = Cone(; rbotMin = rbotMin, rbotMax = rbotMax, rtopMin = rtopMin, rtopMax = rtopMax, φMin = φMin, φMax = φMax, zMin = zMin, zMax = zMax)
 
-function Cone(rbot::R1, rtop::R2, height::H) where {R1<:Real, R2<:Real, H<:Real}
-    T = float(promote_type(R1, R2, H))
-    Cone( T, (T(rbot), T(rtop)), nothing, T(height)/2)
+function surfaces(t::PartialCylinder{T}) where {T}
+    bot_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), -t.hZ), t) 
+    top_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), +t.hZ), t) 
+    mantle = PartialConeMantle{T,:inwards}((t.r, t.r), t.φ, t.hZ, t.origin, t.rotation)
+    e_bot = PartialCircularArea{T}(r = t.r, φ = t.φ, origin = bot_center_pt, rotation = t.rotation)
+    e_top = PartialCircularArea{T}(r = t.r, φ = t.φ, origin = top_center_pt, rotation = -t.rotation * RotZ{T}(π))
+    poly_l = _transform_into_global_coordinate_system(Quadrangle{T}((
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[1], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], -t.hZ)) )), t)
+    poly_r = _transform_into_global_coordinate_system(Quadrangle{T}((
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], +t.hZ)) )), t)
+    # normals of the surfaces show inside the volume primitives. 
+    (e_top, e_bot, mantle, poly_l, poly_r)
 end
 
+####################################################################
+####################################################################
+
+#=
+___
+   \
+    \
+_____\
+
+=#
+const VaryingCylinder{T,CO} = Cone{T,CO,Tuple{T,T},Nothing} # Full in φ
+const PartialVaryingCylinder{T,CO} = Cone{T,CO,Tuple{T,T},Tuple{T,T}}
+
+function _in(pt::CartesianPoint, c::VaryingCylinder{T,ClosedPrimitive}) where {T} 
+    az = abs(pt.z) 
+    (az <= c.hZ || csg_isapprox_lin(az, c.hZ)) && begin
+        r = hypot(pt.x, pt.y) 
+        rz = radius_at_z(c.hZ, c.r[1], c.r[2], pt.z)
+        r <= rz || csg_isapprox_lin(r, rz)
+    end
+end
+function _in(pt::CartesianPoint, c::VaryingCylinder{T,OpenPrimitive}) where {T} 
+    abs(pt.z) < c.hZ &&
+    hypot(pt.x, pt.y) < radius_at_z(c.hZ, c.r[1], c.r[2], pt.z)
+end
```

Comment:
My idea here: why not just define one function and let `csg_isappox_lin` always return `false` when it is an `OpenPrimitive`?

Reply:
Yes, this could be written in one function and also without actually executing the unnecessary parts (like what you mentioned in your comment regarding `_in_angular_interval_closed(::T, ::Nothing)`). However, it would look a bit different and might be  difficult to read. I would suggest to postpone this.